source/tutorials/file-sets.md

```diff
@@ -0,0 +1,661 @@
+(file-sets)=
+# Working with local files
+<!-- TODO: Switch all mentions of unstable to stable once 23.11 is out
+Also remove mention of using unstable
+Also use https://nixos.org/manual/nixpkgs/unstable/#sec-functions-library-fileset
+instead of https://nixos.org/manual/nixpkgs/unstable/#sec-fileset
+-->
+
+To build a local project in a Nix derivation,
+its source files must be accessible
+to the [derivation builder](https://nixos.org/manual/nix/stable/language/derivations#attr-builder).
+Since by default, derivations get built in an [isolated environment](https://nixos.org/manual/nix/stable/command-ref/conf-file.html#conf-sandbox)
+that only allows reading from the Nix store,
+the Nix language has built-in features to copy local files to the store and expose the resulting store paths.
```

Comment:
```suggestion
To build a local project in a Nix derivation, source files must be accessible to its [`builder` executable](https://nixos.org/manual/nix/stable/language/derivations#attr-builder).
Since by default, the `builder` runs in an [isolated environment](https://nixos.org/manual/nix/stable/command-ref/conf-file.html#conf-sandbox) that only allows reading from the Nix store, the Nix language has built-in features to copy local files to the store and expose the resulting store paths.
```

I think it's not wrong to keep these simple mechanisms transparent. I won't fight though, and leave it as an optional suggestion.